framework/include/utils/InputParameters.h

```diff
@@ -996,6 +996,28 @@ InputParameters::setParamHelper<FunctionName, int>(const std::string & /*name*/,
   l_value = oss.str();
 }
 
+template<>
+inline
```

Comment:
Uhm, @permcody, ```inline``` in C++ is not the same thing as ```inline``` in C. This statement has _no_ impact on the executable size as far as I know. It is necessary to keep the full template specialization in the header file. Omit it and you get duplicate symbol errors. I just went with how it was done in that file. In ```MaterialPropertyInterface``` I chose to only forward declare the full specializations in the header (necessary so the compiler knows about them, damn that bug cost me a good half hour) and put the implementations in the source file. Full specializations are not templates anymore. It should improve compile times if we move those into the source file, too, and I'm happy to do that.